src/main/java/net/sf/jabref/logic/groups/GroupDescriptions.java

```diff
@@ -0,0 +1,120 @@
+package net.sf.jabref.logic.groups;
+
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.util.strings.StringUtil;
+import net.sf.jabref.model.groups.ExplicitGroup;
+import net.sf.jabref.model.groups.KeywordGroup;
+import net.sf.jabref.model.groups.SearchGroup;
+
+public class GroupDescriptions {
+
+    public static String getDescriptionForPreview(String field, String expr, boolean caseSensitive, boolean regExp) {
+        String header = regExp ? Localization.lang("This group contains entries whose <b>%0</b> field contains the regular expression <b>%1</b>",
+                field, StringUtil.quoteForHTML(expr))
+                : Localization.lang("This group contains entries whose <b>%0</b> field contains the keyword <b>%1</b>",
```

Comment:
I leave the ternaries as they jointly have some structure, despite each being quite hard to read.